File: activitydemo/src/main/java/com/awesome/activitydemo/Fruit.java
package com.awesome.activitydemo;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Author: JfangZ
 * Email: zhangjingfang@jeejio.com
 * Date: 2021/5/19 10:28
 * Description:
 */
public class Fruit implements Parcelable {
    String name;
    int color;

    public Fruit(String name, int color) {
        this.name = name;
        this.color = color;
    }

    protected Fruit(Parcel in) {
        name = in.readString();
        color = in.readInt();
    }

    public static final Creator<Fruit> CREATOR = new Creator<Fruit>() {
        @Override
        public Fruit createFromParcel(Parcel in) {
            return new Fruit(in);
        }

        @Override
        public Fruit[] newArray(int size) {
            return new Fruit[size];
        }
    };

    @Override
    public String toString() {
        return "Fruit{" +
                "name=" + name +
                ", color=" + color +
                '}';
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeInt(color);
    }
}

File: kotlindemo/src/main/java/com/awesome/kotlindemo/other/Fruit.kt
package com.awesome.kotlindemo.other

/**
 * Author: JfangZ
 * Email: zhangjingfang@jeejio.com
 * Date: 2021/4/2 09:26
 * Description:
 */
// open可被继承。constructor指定主构造函数
open class Fruit constructor() {
    // 主构造函数的初始化代码
    init {
//        println("I'm a Fruit.")
    }

    // 二级构造函数
    constructor(color: String) : this() {
        println("I'm a Father, my color is $color.");
    }

    // 二级构造函数
    constructor(color: String, name: String) : this() {
        println("I'm a Father, I'm a $name and my color is $color.");
    }
}